### PR TITLE
Ec2 specs

### PR DESCRIPTION
**Pull request by vsavinoff**
Introducing the support for Spock tests

### CODE REVIEW

File: src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy [hunk 1]
```diff
@@ -106,10 +108,10 @@ if (canGetOptions(args)) {
 	}
 
 }
+print result
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy]: Debug output should be removed

Comment (imago-storm) [inline src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy]: Please remove debug outputs here

File: src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy [hunk 2]
```diff
@@ -125,11 +127,36 @@ def loginEC2(credential, serviceURL) {
...
+
+	def ec2 = new AmazonEC2Client(awsCreds, clientConfig)
+	ec2.setEndpoint(serviceURL)
+
+	ec2
+}
+
 boolean canGetOptions(args) {
+
+    // println "SPARTA"
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy]: Cleanup here is required

Comment (imago-storm) [inline src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy]: And here too

File: src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy [hunk 3]
```diff
@@ -229,3 +256,52 @@ def getAvailabilityZones(AmazonEC2Client ec2) {
...
+    if (args.credential && isCollectionOrArray(args.credential) && args.credential.size() > 0) {
+        for (c in args.credential) {
+            if (!(c.credentialName =~ '_proxy_credential$')) {
+                return c;
+            }
+        }
+    }
+    return null;
+}
+// def getAWSCredential
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/parameterOptions/ec2Options.groovy]: Cleanup required.

File: src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy [hunk 4]
```diff
@@ -87,7 +89,12 @@ if (canValidate(args)) {
 
 //--------------------Helper functions-----------------------------//
 
+// TODO: Switch validation from credential[0] to getAWSCredential
 boolean canValidate(args) {
+    print "CANVALIDATE2";
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy]: Cleanup required

File: src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy [hunk 5]
```diff
@@ -96,9 +103,55 @@ boolean canValidate(args) {
...
+        rv.host = vals[0][1];
+        rv.port = vals[0][2] as int;
+    }
+    return rv;
+}
+
 def doValidations(args) {
-	def credential = args.credential[0]
-	def parameters = args.parameters
+    print "DOVALIDATIONS"
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy]: Cleanup required.

File: src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy [hunk 6]
```diff
@@ -96,9 +103,55 @@ boolean canValidate(args) {
...
+    print args.credential
```
Comment (justnoxx) [inline src/main/resources/project/procedures/form_scripts/validation/createConfiguration.groovy]: Same here.

**Pull request merged**